Mathlib/Analysis/Normed/Module/WeakDual.lean

```diff
@@ -255,4 +256,56 @@ theorem isCompact_closedBall [ProperSpace 𝕜] (x' : StrongDual 𝕜 E) (r : 
     IsCompact (toStrongDual ⁻¹' closedBall x' r) :=
   isCompact_of_bounded_of_closed isBounded_closedBall (isClosed_closedBall x' r)
 
+variable (𝕜 V : Type*) [NontriviallyNormedField 𝕜] [SeminormedAddCommGroup V] [NormedSpace 𝕜 V]
+variable [TopologicalSpace.SeparableSpace V] (K : Set (WeakDual 𝕜 V))
+
+/- There exists a sequence of continuous functions that separates points on the weak dual. -/
+lemma exists_continuous_separating : ∃ (gs : ℕ → (WeakDual 𝕜 V) → 𝕜),
+    (∀ n, Continuous (gs n)) ∧ (∀ ⦃x y⦄, x≠y → ∃ n, gs n x ≠ gs n y) := by
+  set vs := TopologicalSpace.denseSeq V
+  use (fun n ↦ fun ϕ ↦ (ϕ : WeakDual 𝕜 V) (vs n))
+  constructor
+  · exact fun n ↦ WeakDual.eval_continuous (vs n)
+  · intro w y w_ne_y
+    contrapose! w_ne_y
+    have : Set.EqOn w y (.range vs) := by
+      simpa [Set.eqOn_range] using (Set.eqOn_univ (w ∘ vs) (y ∘ vs)).mp fun x _ ↦ w_ne_y x
+    exact DFunLike.coe_fn_eq.mp (Continuous.ext_on (TopologicalSpace.denseRange_denseSeq V)
+      (map_continuous w) (map_continuous y) this)
```

Comment:
minor clean-up. Note the `grind` call. You'll need to `open TopologicalSpace` somewhere, preferably prior to your `SeparableSpace` variable above.
```suggestion
  use (fun n φ ↦ φ (denseSeq V n))
  constructor
  · exact fun _ ↦ WeakDual.eval_continuous _
  · intro w y w_ne_y
    contrapose! w_ne_y
    exact DFunLike.ext'_iff.mpr <| (map_continuous w).ext_on
      (denseRange_denseSeq V) (map_continuous y) (by grind [Set.eqOn_range])
```